### DISCUSSION TITLE
Is there a hook for project changed?

### DISCUSSION THREAD

**Discussion by 616b2f** [GnikDroy/projections.nvim#41] [Q&A] [answered]
What I try todo is to call some functions when I select a project via telescope.
I saw that there are hooks for session management, but I guess it's not what I search for.

Many thanks in advance.

**Comment by GnikDroy** [comment, marked as answer]
There is no such hook. 

You can hook into switcher yourself.

```lua
local switcher = require("projections.switcher")
local original_switch_function = switcher.switch

switcher.switch = function(...)
    -- pre hooks here
    local result = original_switch_function(...)
    -- unconditional post hooks here
    if result then
        --- post hook that only runs if project switching was successful
    end
    return result
end
```

**Reply by GnikDroy** [reply]
Additionally, if you only care about the telescope UI. You can pass in `action` optional argument into the picker, to perform any action you want.

More information about this is in Telescope docs.
 
```lua
-- The telescope UI will print details about the selection
require('telescope').load_extension('projections')
vim.keymap.set("n", "<leader>fp", function()
    local find_projects = require("telescope").extensions.projections.projections
    find_projects({ action = function(selected)
        print(vim.inspect(selected))
    end})
end)
 ```
 
 The default action function used by projections is:
 https://github.com/GnikDroy/projections.nvim/blob/889914169fa1f5c00fb8659653b5a8330fb223e6/lua/telescope/_extensions/projections.lua#L66-L70

**Reply by 616b2f** [reply]
Thank you! This was a fast response :), I will try them out.
